Shared/Samples/Find route in mobile map package/FindRouteInMobileMapPackageView.swift

```diff
@@ -0,0 +1,204 @@
+            )  else { return }
+            mapPackages = await loadMapPackages(from: bundleMapPackageURLs)
+        }
+        .task(id: importedFileURLs) {
+            // Load the new mobile map packages when file URLs are imported.
+            guard let importedFileURLs else { return }
+            
+            let newMapPackages = await loadMapPackages(from: importedFileURLs)
+            mapPackages.append(contentsOf: newMapPackages)
+            self.importedFileURLs = nil
+        }
+        .alert(isPresented: $errorAlertIsShowing, presentingError: error)
+    }
+    
+    /// Loads a list of mobile map packages from a given list of URLs.
+    /// - Parameter URLs: The list of file URLs pointing to "mmpk" files.
+    /// - Returns: A list of loaded mobile map packages.
+    private func loadMapPackages(from URLs: [URL]) async -> [MobileMapPackage] {
+        do {
+            // Create and load a mobile map package using each url.
+            let mapPackages = try await withThrowingTaskGroup(of: MobileMapPackage.self) { group in
+                for url in URLs {
+                    group.addTask {
+                        let mapPackage = MobileMapPackage(fileURL: url)
+                        try await mapPackage.load()
+                        return mapPackage
+                    }
+                }
+                var loadedMapPackages: [MobileMapPackage] = []
+                for try await loadedMapPackage in group {
+                    loadedMapPackages.append(loadedMapPackage)
+                }
+                return loadedMapPackages.sorted { $0.item?.name ?? "" < $1.item?.name ?? "" }
+            }
+            
+            return mapPackages
+        } catch {
+            self.error = error
+            return []
+        }
+    }
+}
+
+private extension FindRouteInMobileMapPackageView {
+    /// A list of the maps in a given map package.
+    struct MobileMapListView: View {
+        /// The mobile map package containing the maps.
+        let mapPackage: MobileMapPackage
+        
+        var body: some View {
+            // Create a list row for each map in the map package.
+            ForEach(mapPackage.maps.enumeratedArray(), id: \.offset) { (offset, map) in
+                let mapName = map.item?.name.titleCased ?? "Map \(offset + 1)"
+                
+                // The navigation link to the map.
+                NavigationLink {
+                    Group {
+                        if let locatorTask = mapPackage.locatorTask {
+                            MobileMapView(map: map, locatorTask: locatorTask)
+                        } else {
+                            MapView(map: map)
+                        }
+                    }
+                    .navigationTitle(mapName)
+                } label: {
+                    HStack {
+                        // The image of the map for the row.
+                        Image(uiImage: map.item?.thumbnail?.image ?? UIImage())
```

Comment:
Maybe a question mark if the image can't be resolved might be good.

```suggestion
                        Image(uiImage: map.item?.thumbnail?.image ?? UIImage(systemName: "questionmark")!)
```